bundles/org.openhab.core.model.yaml/src/main/java/org/openhab/core/model/yaml/internal/things/YamlThingProvider.java

```diff
@@ -176,7 +175,7 @@ public void deactivate() {
 
     @Override
     public Collection<Thing> getAll() {
-        return thingsMap.values().stream().flatMap(list -> list.stream()).collect(Collectors.toList());
+        return thingsMap.values().stream().flatMap(list -> list.stream()).toList();
```

Comment:
I know both are not equivalent but I am never comfortable with what option should be used.
Are you sure of this change is correct in our context and if it is can you explain why please ?

Reply:
toList returns an unmodifiable list, whereas collect(Collectors.toList()) returns a modifiable one, although afaik, not guaranteed.

I am just following this: https://github.com/openhab/openhab-core/pull/3831

Many existing implementations of getAll() in core would return a Stream.toList().

In short, use Stream.toList() whenever possible. For me it's just shorter to type and read.